core/templates/pages/contributor-dashboard-admin-page/contributor-admin-dashboard-page.component.spec.ts

```diff
@@ -350,19 +342,152 @@ describe('Contributor dashboard Admin page', () => {
+
+        component.selectedFirstActivity = new Date(
+          new Date().getTime() + 24 * 60 * 60 * 1000
+        );
+        component.selectFirstActivity();
+        expect(component.selectedFirstActivityInDays).toEqual(0);
+      })
+    );
+
+    it(
+      'should not update first activity in days when selected first' +
+        ' date is less than selected last date',
+      fakeAsync(() => {
+        component.ngOnInit();
+        tick();
+        fixture.detectChanges();
+
+        component.selectedFirstActivity = new Date(
+          new Date().getTime() - 24 * 60 * 60 * 1000 * 91
+        );
+        component.selectFirstActivity();
+        expect(component.selectedFirstActivityInDays).toEqual(0);
+      })
+    );
+
+    it('should set default last activity at 90 days', fakeAsync(() => {
+      component.ngOnInit();
+      tick();
+      fixture.detectChanges();
+
+      expect(component.selectedLastActivityInDays).toEqual(90);
+    }));
+
+    it(
+      'should update last activity in days via dateChange' + ' event',
+      fakeAsync(() => {
+        component.ngOnInit();
+        tick();
+        fixture.detectChanges();
+
+        const newDate: Date = new Date(
+          new Date().getTime() - 24 * 60 * 60 * 1000 * 7
+        );
+        component.changeLastActivity(newDate);
+        expect(component.selectedLastActivityInDays).toEqual(7);
+      })
+    );
+
+    it(
+      'should update last activity in days when selected last' +
+        ' date is valid',
+      fakeAsync(() => {
+        component.ngOnInit();
+        tick();
+        fixture.detectChanges();
+
+        component.selectedLastActivity = new Date(
+          new Date().getTime() - 24 * 60 * 60 * 1000 * 30
+        );
+        component.selectLastActivity();
+        expect(component.selectedLastActivityInDays).toEqual(30);
+      })
+    );
+
+    it(
+      'should not update last activity in days when selected last' +
+        ' date is more than today',
+      fakeAsync(() => {
+        component.ngOnInit();
+        tick();
+        fixture.detectChanges();
+
+        component.selectedLastActivity = new Date(
+          new Date().getTime() + 24 * 60 * 60 * 1000
+        );
+        component.selectLastActivity();
+        expect(component.selectedLastActivityInDays).toEqual(90);
+      })
+    );
+
+    it(
+      'should not update last activity in days when selected last' +
+        ' date is more than selected first date',
+      fakeAsync(() => {
+        component.ngOnInit();
+        tick();
+        fixture.detectChanges();
+
+        component.selectedFirstActivity = new Date(
+          new Date().getTime() - 24 * 60 * 60 * 1000 * 30
+        );
+        component.selectFirstActivity();
+        component.selectedLastActivity = new Date(
+          new Date().getTime() - 24 * 60 * 60 * 1000 * 29
+        );
+        component.selectLastActivity();
+        expect(component.selectedLastActivityInDays).toEqual(90);
+      })
+    );
```

Comment:
Refactor these similar to the tests suggested in the above comments.

Reply:
done